sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/MetricData.cs

```diff
@@ -27,12 +27,28 @@ public MetricsData(int version, Metric metric, MetricPoint metricPoint) : base(v
             switch (metric.MetricType)
             {
                 case MetricType.DoubleSum:
-                    metricDataPoint = new MetricDataPoint(metric.Name, metricPoint.GetSumDouble());
-                    metricDataPoint.DataPointType = DataPointType.Aggregation;
+                    metricDataPoint = new MetricDataPoint(metric.Name, metricPoint.GetSumDouble())
+                    {
+                        DataPointType = DataPointType.Aggregation
+                    };
                     break;
                 case MetricType.DoubleGauge:
-                    metricDataPoint = new MetricDataPoint(metric.Name, metricPoint.GetGaugeLastValueDouble());
-                    metricDataPoint.DataPointType = DataPointType.Measurement;
+                    metricDataPoint = new MetricDataPoint(metric.Name, metricPoint.GetGaugeLastValueDouble())
+                    {
+                        DataPointType = DataPointType.Measurement
+                    };
+                    break;
+                case MetricType.LongSum:
+                    metricDataPoint = new MetricDataPoint(metric.Name, metricPoint.GetSumLong())
```

Comment:
maybe add a link to https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/numeric-conversions#implicit-numeric-conversions .

Reply:
We have a PR pending in draft state - https://github.com/Azure/azure-sdk-for-net/pull/27544. As the implicit or explicit conversion from long to double loses a precision.

Reply:
That PR is doing a different (related) thing. Its trying to store count(long) into an int value..
This one is accepting long, and using the built-in conversion to double. It does lose precision, hence i suggested add comment to link to docs, which talks about prevision lose..

Reply:
Added a comment and link for reference.